rubix-common/src/main/java/com/qubole/rubix/common/utils/ClusterUtil.java

```diff
@@ -50,4 +64,66 @@ public static String getMasterHostname(Configuration conf)
     log.debug("No hostname found in etc/*-site.xml, returning localhost");
     return "localhost";
   }
+
+  public static synchronized Configuration applyRubixSiteConfig(Configuration conf) throws Exception
+  {
+    if (rubixSiteConfiguration == null) {
+      rubixSiteConfiguration  = new HashMap<>();
+      String configFileLocation = CacheConfig.getRubixSiteLocation(conf);
+      File configFile = new File(configFileLocation);
+      if (!configFile.exists()) {
+        log.warn(String.format("Config file does not exist on location: %s", configFileLocation));
+        if (!configFileLocation.equals(DEFAULT_RUBIX_SITE_LOCATION)) {
+          throw new IOException("Rubix config file not found on path: " + configFileLocation);
+        }
+      } else {
+        rubixSiteConfiguration = readXML(configFile);
+      }
+    }
+
+    if (!CacheConfig.isRubixSiteConfigApplied(conf)) {
+      for (Map.Entry<String, String> rubixConfig : rubixSiteConfiguration.entrySet()) {
```

Comment:
We can just depend on Configuration class' methods, how about something like this:
```
  private static volatile AtomicReference<Boolean> rubixSiteExists = new AtomicReference<>();

  public static synchronized Configuration applyRubixSiteConfig(Configuration conf) throws Exception
  {
    if (rubixSiteExists.get() != null && !rubixSiteExists.get()) {
      return conf;
    }

    if (CacheConfig.isRubixSiteConfigApplied(conf)) {
      return conf;
    }

    File configFile = null;
    if (rubixSiteExists.get() != null) {
      synchronized (rubixSiteExists) {
        if (rubixSiteExists.get() != null) {
          String configFileLocation = CacheConfig.getRubixSiteLocation(conf);
          configFile = new File(configFileLocation);
          if (configFile.exists() && configFile.isFile() && configFile.length() != 0) {
            rubixSiteExists.set(true);
          }
          else {
            rubixSiteExists.set(false);
            return conf;
          }
        }
      }
    }

    conf.addResource(configFile.toURL());
    CacheConfig.setRubixConfigApplied(conf, true);
    return conf;
  }
```